pkg/asset/imagebased/image/baseiso.go

```diff
@@ -0,0 +1,140 @@
+package image
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/coreos/stream-metadata-go/stream"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/openshift/installer/pkg/rhcos/cache"
+	"github.com/openshift/installer/pkg/types"
+)
+
+var (
+	baseIsoFilename = ""
+
+	defaultCoreOSStreamGetter = rhcos.FetchCoreOSBuild
+)
+
+// BaseIso generates the base ISO file for the image.
+type BaseIso struct {
+	File *asset.File
+
+	streamGetter CoreOSBuildFetcher
+}
+
+// CoreOSBuildFetcher will be to used to switch the source of the coreos metadata.
+type CoreOSBuildFetcher func(ctx context.Context) (*stream.Stream, error)
+
+var _ asset.WritableAsset = (*BaseIso)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (i *BaseIso) Name() string {
+	return "BaseIso Image"
+}
+
+// Dependencies returns dependencies used by the asset.
+func (i *BaseIso) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the base ISO.
+func (i *BaseIso) Generate(dependencies asset.Parents) error {
+	var err error
+	var baseIsoFileName string
+
+	if urlOverride, ok := os.LookupEnv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); ok && urlOverride != "" {
```

Comment:
Why?
Is this something you got from the agent iso?

Reply:
Yes, it is indeed how the agent behaves. I also thought that since the installer supports this override, users might expect us to support it as well. Although it is anyway not advised for a user to do so...

Of course, we don't have to support this override (or at least not to begin with). Are you thinking of any {up/down}sides?